app/src/components/GrantOwnerCard.vue

```diff
@@ -0,0 +1,116 @@
+        <div class="mb-2">
+          <span class="text-grey-400">Donations:</span>
+          <span @click="pushRoute({ name: 'dgrants-id', params: { id: id.toString() } })" class="ml-1 link">{{
+            grantInfo.contributionCount
+          }}</span>
+        </div>
+
+        <!-- Status -->
+        <div class="mt-10">
+          <!-- approved (green)-->
+          <div v-if="status == 'approved'">
+            <span class="px-8 py-4 font-medium bg-teal">Approved</span>
+          </div>
+
+          <!-- pending (grey)-->
+          <div v-else>
+            <span class="px-8 py-4 font-medium bg-grey-100">Pending</span>
+          </div>
+        </div>
+      </div>
+    </article>
+
+    <!--------- INTERACTION BAR - FOR EDIT, CLAIM, HIDE, ETC ... ---------->
+    <article class="py-8 mt-8">
+      <div class="flex flex-wrap gap-x-10 gap-y-4 justify-end">
+        <!--EDIT-->
+        <div @click="pushRoute({ name: 'dgrants-id', params: { id: id.toString() }, query: { edit: 'true' } })">
+          <div class="flex items-center gap-x-2 cursor-pointer group">
+            <EditIcon class="icon icon-primary icon-small" />
+            <span class="text-grey-400 group-hover:text-grey-500">Edit</span>
+          </div>
+        </div>
+      </div>
+    </article>
+  </section>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref } from 'vue';
+// --- Methods and Data ---
+import { filterContributionsByGrantId } from 'src/utils/data/contributions';
+import { formatNumber, pushRoute } from 'src/utils/utils';
+
+// --- Components/icons ---
+import useDataStore from 'src/store/data';
+import { Edit3Icon as EditIcon } from '@fusion-icons/vue/interface';
+
+function useGrantInfo(id: number) {
+  const { grantRounds, grantContributions, approvedGrantsPk } = useDataStore();
+
+  const grantId = ref<number>(id);
+
+  const status = computed(() => {
+    if (approvedGrantsPk && approvedGrantsPk.value) {
+      return approvedGrantsPk.value.includes(id) ? 'approved' : 'pending';
+    }
+    console.error('could not compute approval status of grant');
+    return 'pending';
+  });
+
+  function getGrantInfo(grantId: number) {
+    const contributions = filterContributionsByGrantId(grantId, grantContributions?.value || []);
+    const contributionCount = contributions.length;
+    const raised = `${formatNumber(
+      contributions.reduce((total, contribution) => contribution?.amount + total, 0),
+      2
+    )} ${grantRounds.value && grantRounds.value[0] && grantRounds.value[0].donationToken.symbol}`;
+
+    return { raised: raised, contributionCount: contributionCount };
+  }
+
+  const grantInfo = computed(() => getGrantInfo(grantId.value));
+
+  return { status, grantInfo };
+}
+
+export default defineComponent({
+  name: 'GrantOwnerCard',
+  components: { EditIcon },
+  props: {
+    id: { type: Number, required: true },
+    name: { type: String, required: true },
+    imgurl: { type: String, required: true },
```

Comment:
This should take `logoPtr` as type `MetaPtr` as a prop, and resolve it to an image URL. We always want to pass around the full pointer object and resolve it as late as possible. I think this PR needs to be rebased against `main` because you'll see we removed all `imgurl` props from components and now pass in the logo prop like this: https://github.com/dcgtc/dgrants/blob/a752c90bc1dbc44a901d22e3efe05557f0726361/app/src/components/GrantCard.vue#L84

And resolve it as late as possible like this: https://github.com/dcgtc/dgrants/blob/a752c90bc1dbc44a901d22e3efe05557f0726361/app/src/components/GrantCard.vue#L18-L21

Reply:
Hmm the parent was actually taking care of that but i see what you're saying. this is cleaner